appimaged broken, likely by #115

Hello @CalebQ42, thanks for your PR https://github.com/probonopd/go-appimage/pull/115 which I had merged. Did you try the resulting appimaged AppImage on a Ubuntu Live ISO? 

Looks like there is something going wrong badly:

```
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaHdLg4R type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaTcQia5 type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaLBnqdi type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaZLYViv type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaSa1XkI type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimavFc423 type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaaE716g type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appima9QUqbu type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimamavYCK type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaDuKzYT type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaAtUyha type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaalNw2C type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaZDKb5P type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appima7zWe02 type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaNGq55f type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimasmNvOJ type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaP3WiOU type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimair0eT7 type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appima7hBeUk type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaH7ByZx type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaelYE4K type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimacguQ9X type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimajM5cdb type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaBZHkeo type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appima9FmzZd type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appiman5cdQn type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaFkhM0D type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaWc0x4Q type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appima3t2Bf5 type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimantZtch type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimakBKGnv type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaueIBkH type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaqo7XmU type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaetWSs7 type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaSnbXBl type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaQCksCx type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimamDJhNL type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimalRnWOY type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaq8haVb type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaGiD0Wo type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaN3dMUA type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appima9vkk6O type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimanBZxb2 type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appima6Ohz8G type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimawKVeJQ type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimabzIikk type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appima8Lnpox type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appima2rq7rK type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaHFd30T type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaIE1Iza type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaefw4ak type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaTEiBHA type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimajCsViK type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimag8WJO0 type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaKdQTqa type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimajn3rWq type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaSnRA1D type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimam0AEBN type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaE0HbH0 type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaMqXNKd type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimastAXOq type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaNCZmTD type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appima7my8oU type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaZkBien type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaNqp3iA type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaNCRCmN type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimayeO4u0 type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimayDW5wd type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaqOREAq type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaDWEbED type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimajpJ3JQ type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaR1PsI3 type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaxcyGPt type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appima3ks4VG type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimascr4WT type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaYmGa86 type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appima7jjsbk type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimavRg69w type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimamp6YgK type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimawXq2jX type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appima82a6ma type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaGE5jNm type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimauMNJds type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimavDDtmI type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaIzgYFn type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaou6AWS type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimasmKzK5 type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appima9ldmKy type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaii9IRL type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimajChQSY type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimajRTHWb type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimalQ42aB type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimasSEWPK type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaQUk4j1 type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimabgUYke type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaJHGEor type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaDXiAtE type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimakih8yR type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimatwPvlh type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimabWy3Ju type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appima8npuUH type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimazihCRU type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaMiMqT7 type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaQ7Qs1k type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appima9AoP5x type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaJ09LbL type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appima4hwDbY type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appima19E17a type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appima1DMCgo type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaPWj3nB type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaFY2Efl type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appima0xyuju type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaxMOwxH type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaet4ZuU type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimajLEQw8 type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appiman7UpCk type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appima6bDUHx type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimasJrZFK type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaTby8KX type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaHggOQa type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaUbbdXn type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaN6AN1A type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appimaDcH6WO type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
appimaged-635-x86_64.AppImage on /tmp/.mount_appima40jv80 type fuse.appimaged-635-x86_64.AppImage (ro,nosuid,nodev,relatime,user_id=999,group_id=999)
```

```
monitor: Monitoring DBus session bus
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x82a280]

goroutine 1 [running]:
github.com/CalebQ42/squashfs.(*fileReader).Close(0x0, 0xb290e0, 0x0)
	github.com/CalebQ42/squashfs@v0.3.6/filereader.go:62 +0x20
github.com/CalebQ42/squashfs.(*File).Close(0xc000072320, 0xb290a0, 0xc000072320)
	github.com/CalebQ42/squashfs@v0.3.6/file.go:490 +0xd9
gopkg.in/ini%2ev1.(*File).reload(0xc00025e000, 0xb29760, 0xc0001d6060, 0xb29040, 0xc0001d6310)
	gopkg.in/ini.v1@v1.62.0/file.go:291 +0xf6
gopkg.in/ini%2ev1.(*File).Reload(0xc00025e000, 0xc00025e000, 0xb29760)
	gopkg.in/ini.v1@v1.62.0/file.go:297 +0x72
gopkg.in/ini%2ev1.LoadSources(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	gopkg.in/ini.v1@v1.62.0/ini.go:147 +0x2fc
gopkg.in/ini%2ev1.Load(...)
	gopkg.in/ini.v1@v1.62.0/ini.go:157
github.com/probonopd/go-appimage/src/goappimage.NewAppImage(0xc00053c200, 0x3f, 0xc00053c200, 0xc0001ddf80, 0xc00015c400)
	github.com/probonopd/go-appimage/src/goappimage/appimage.go:71 +0x325
main.NewAppImage(0xc00053c200, 0x3f, 0x17, 0xa76c26, 0x1)
	github.com/probonopd/go-appimage/src/appimaged/appimage.go:47 +0x70
main.watchDirectoriesReally(0xc0001ce400, 0x6, 0x8)
	github.com/probonopd/go-appimage/src/appimaged/appimaged.go:456 +0x256
main.watchDirectories()
	github.com/probonopd/go-appimage/src/appimaged/appimaged.go:432 +0x725
main.main()
	github.com/probonopd/go-appimage/src/appimaged/appimaged.go:199 +0x58a
```

--- a/go.mod
+++ b/go.mod
@@ -3,7 +3,7 @@ module github.com/probonopd/go-appimage
 go 1.13
 
 require (
-	github.com/CalebQ42/squashfs v0.3.6
+	github.com/CalebQ42/squashfs v0.3.7
 	github.com/StackExchange/wmi v0.0.0-20190523213315-cbe66965904d // indirect
 	github.com/acobaugh/osrelease v0.0.0-20181218015638-a93a0a55a249
 	github.com/adrg/xdg v0.2.3

--- a/src/appimaged/desktop.go
+++ b/src/appimaged/desktop.go
@@ -241,7 +241,7 @@ func fixDesktopFile(path string) error {
 		output = bytes.Replace(input, []byte("=`"), []byte("="), -1)
 		output = bytes.Replace(output, []byte("`\n"), []byte("\n"), -1)
 	}
-	output = bytes.ReplaceAll(output, []byte(","), []byte(";"))
+	output = bytes.ReplaceAll(output, []byte("；"), []byte(";"))
 
 	if err = ioutil.WriteFile(path, output, 0755); err != nil {
 		return err

--- a/src/appimaged/desktop.go
+++ b/src/appimaged/desktop.go
@@ -240,10 +240,8 @@ func fixDesktopFile(path string) error {
 	if bytes.Contains(input, []byte("=`")) {
 		output = bytes.Replace(input, []byte("=`"), []byte("="), -1)
 		output = bytes.Replace(output, []byte("`\n"), []byte("\n"), -1)
-		//some issues that appear now that we're using the original .desktop file
-		output = bytes.Replace(output, []byte("\n;\n"), []byte("\n"), -1)
-		output = bytes.Replace(output, []byte("\n; \n"), []byte("\n"), -1)
 	}
+	output = bytes.ReplaceAll(output, []byte(","), []byte(";"))
 
 	if err = ioutil.WriteFile(path, output, 0755); err != nil {
 		return err

--- a/src/goappimage/appimage.go
+++ b/src/goappimage/appimage.go
@@ -77,7 +77,7 @@ func NewAppImage(path string) (*AppImage, error) {
 		var line string
 		line, err = buf.ReadString('\n')
 		if strings.Contains(line, ";") {
-			line = strings.ReplaceAll(line, ";", ",")
+			line = strings.ReplaceAll(line, ";", "；") //replacing it with a fullwidth semicolon (unicode FF1B)
 		}
 		desktop = append(desktop, line...)
 	}

--- a/src/goappimage/appimage.go
+++ b/src/goappimage/appimage.go
@@ -1,6 +1,7 @@
 package goappimage
 
 import (
+	"bufio"
 	"bytes"
 	"errors"
 	"io"
@@ -68,7 +69,20 @@ func NewAppImage(path string) (*AppImage, error) {
 	if err != nil {
 		return nil, err
 	}
-	ai.Desktop, err = ini.Load(desktopFil)
+
+	//cleaning the desktop file so it can be parsed properly
+	var desktop []byte
+	buf := bufio.NewReader(desktopFil)
+	for err == nil {
+		var line string
+		line, err = buf.ReadString('\n')
+		if strings.Contains(line, ";") {
+			line = strings.ReplaceAll(line, ";", ",")
+		}
+		desktop = append(desktop, line...)
+	}
+
+	ai.Desktop, err = ini.Load(desktop)
 	if err == nil {
 		ai.Name = ai.Desktop.Section("Desktop Entry").Key("Name").Value()
 	}

--- a/src/goappimage/archivereader.go
+++ b/src/goappimage/archivereader.go
@@ -209,7 +209,7 @@ func (r *type2Reader) FileReader(filepath string) (io.ReadCloser, error) {
 		if fil.IsDir() {
 			return nil, errors.New("Path is a directory: " + filepath)
 		}
-		return fil, nil
+		return ioutil.NopCloser(fil), nil
 	}
 	filepath, err := r.cleanPath(filepath)
 	filepath = r.SymlinkPathRecursive(filepath)

